### ISSUE TITLE
Checksum mismatch in go.mod

### ISSUE THREAD

**Issue by lyspj** [dependabot/dependabot-core#3073]
## Report
**Package manager/ecosystem**
go_modules

**What you expected to see, versus what you actually saw**
I expected Dependabot to resolve my go.mod dependency file. 

Instead I got the following message:
>Dependabot can't resolve your Go dependency files.
>
>As a result, Dependabot couldn't update your dependencies.
>
>The error Dependabot encountered was:
>verifying github.com/GoogleCloudPlatform/berglas@v0.5.3/go.mod: checksum mismatch
>     downloaded: h1:uTiStG2vdRf3SZm7ed4aF1ydgOB2b+8zfYgk+quSdLo=
>     go.sum:     h1:2Jj1E+vIaeqcay89b4yMNHDSOveBl5odI/Lws6t9PnQ=
>
>SECURITY ERROR
>This download does NOT match an earlier download recorded in go.sum.
>The bits may have been replaced on the origin server, or an attacker may
>have intercepted the download attempt.
>
>For more information, see 'go help module-auth'.

I believe that the value in `go.sum` is correct. 

## Steps to verify checksum
To see what checksum a fresh download of Berglas@0.5.3 has, I carried out the following steps (I guess there might be a golang command makes this easier? Please let me know if there's a better way :-) )

1. Run `go clean --modcache`
2. Put the following in a file called main.go:
```
package main

import (
	"fmt"
	"github.com/GoogleCloudPlatform/berglas/pkg/berglas"
)

func main() {
	if berglas.IsReference("abc") {
		fmt.Println("Hello")
	}
}
```
3. Run `go mod init example.com/m/v2`
4. Run `go mod tidy`

Running `go mod verify` results in `all modules verified`, and `go.sum` contains the following two lines (amongst others):

> github.com/GoogleCloudPlatform/berglas v0.5.3 h1:9ZW2xaNDC5Yhop51OFKSM/Vk3UAKcWxd2oiqwEz+8R0=
> github.com/GoogleCloudPlatform/berglas v0.5.3/go.mod h1:2Jj1E+vIaeqcay89b4yMNHDSOveBl5odI/Lws6t9PnQ=

In addition, the checksum shown in one of the Berglas example projects matches the one above: https://github.com/GoogleCloudPlatform/berglas/blob/main/examples/kubernetes/go.sum

## Conclusion 

It _appears_ that the `berglas@0.5.3` checksum in my version of go.sum is correct (or, at least, consistent with that in other projects), and the checksum for the version downloaded by Dependabot is different. I'm not sure why that might be?

**Comment by davidsbond** [comment]
I've also experience this issue several times with various dependencies. It prevents any pull requests from being opened by dependabot. 

Currently it's erroring with:

```
verifying github.com/google/flatbuffers@v1.12.0: checksum mismatch
```

**Comment by markliederbach** [comment]
Similar issue here. For me, it's happening on:
```
verifying github.com/vartanbeno/go-reddit/v2@v2.0.0: checksum mismatch
```

**Comment by case** [comment]
We saw this earlier today as well.

<img width="883" alt="Screen Shot 2021-11-04 at 07 00 20" src="https://user-images.githubusercontent.com/20105/140377342-3e58202c-10db-49e3-b197-4e3bc6d1a797.png">

But:

```
$ go mod verify
all modules verified
```

It seems transient though; I just checked again a few mins ago, and now the error is gone.

**Comment by case** [comment]
And the error is back this morning:

> Dependabot encountered the following error:
> 
> verifying `github.com/envoyproxy/go-control-plane@v0.10.0`: checksum mismatch

Anyone else seeing this?

**Comment by rashedkvm** [comment]
I am seeing the same error for the past few days. 

> Dependabot encountered the following error:
verifying github.com/vmware-tanzu/carvel-imgpkg@v0.29.0: checksum mismatch

**Comment by jerbob92** [comment]
I _thought_ I knew how to fix it, but it appears I was wrong, I have it again:
```
verifying github.com/docker/distribution@v2.8.0+incompatible: checksum mismatch verifying github.com/docker/distribution@v2.8.0+incompatible: checksum mismatch verifying github.com/docker/distribution@v2.8.0+incompatible: checksum mismatch
```

This time it's not in the first require group.

**Comment by jeffwidman** [comment]
Historically, there were a couple of edge cases bugs in `go` that could cause this.

Nowadays, this is _generally_ an issue with the package maintainer pushing a bad release.

I spot-checked a couple of examples in this thread:
* `github.com/docker/distribution@v2.8.0+incompatible` was a package problem, which is why they released `v2.8.1` so fast: https://github.com/distribution/distribution/pull/3585
* `github.com/go-openapi/spec@v0.20.5` I'm guessing was a package issue given that there was no commit diff... most likely they just re-pushed a new tag with a working checksum https://github.com/go-openapi/spec/compare/v0.20.5...v0.20.6

This can also happen if you get something screwed up locally... I worked primarily in `go` for several years before joining the dependabot team and very occasionally saw this. One time a colleague managed to trace it to very infrequent and subtle data corruption in the corporate VPN server.

My suspicion is this _could_ (and probably _does_) happen very infrequently in our internal infra for similar flaky things that are super rare. Given that we're not getting consistent reports of this, and when it does happen it's almost always a problem with either the package itself or the user's computer, I'm going to close.

If you do start seeing this repeatedly, please file a new issue and we can dig in a little more to doublecheck whether it's something here in core or not.
